applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -520,29 +554,16 @@ static void data_send(void)
 		/* This error might occurs when data has not been obtained prior
 		 * to data encoding.
 		 */
-		LOG_DBG("No new data to encode, error: %d", err);
-		return;
-	} else if (err) {
+		LOG_DBG("No new data to encode");
+	} else if (err == 0) {
```

Comment:
I think I'd stick with the old flow and check `if (err)` and return if there is an error instead of checking for `err == 0`. Just a tad easier to read the code IMO

Reply:
Problem is that I don't want to return for ENODATA and ENOTSUP. And, I don't want to call `data_send` in those case either. I'm open to suggestions.
How about a switch case?

```	switch (err) {
	case -ENODATA:
		/* Neighbor cell data encoding not supported */
		break;
	case -ENOTSUP:
		LOG_DBG("No neighbor cells data to encode, error: %d", err);
		break;
	case 0:
		LOG_DBG("Neighbor cell data encoded successfully");
		data_send(DATA_EVT_NEIGHBOR_CELLS_DATA_SEND, NEIGHBOR_CELLS, &codec);
		break;
	default:
		LOG_ERR("Error encoding neighbor cells data: %d", err);
		SEND_ERROR(data, DATA_EVT_ERROR, err);
		return;
	}
```

Reply:
@jtguggedal

Reply:
Yeah, a switch case looks nice. I'd start with the 0 case